allennlp/training/trainer.py

```diff
@@ -1048,6 +1055,13 @@ def from_partial_objects(
         If you're not using `FromParams`, you can just construct these arguments in the right order
         yourself in your code and call the constructor directly.
         """
+        if cuda_device is None:
```

Comment:
Unfortunate that this has to be in both places, but it looks like that's necessary.  It might work to put this into a shared method or something, though, like the existing `check_for_gpu` or `int_to_device`.

Reply:
I'll do that the third time I need it :-)